corgidrp/l1_to_l2a.py

```diff
@@ -49,24 +51,39 @@ def prescan_biassub(input_dataset, noise_maps=None, return_full_frame=False, det
                 raise Exception(f"Observation type of frame {i} is not 'SCI' or 'ENG' or 'ENG_EM' or 'EMG_CONV'")
 
         # Get the reliable prescan area
-        prescan = slice_section(frame_data, obstype, 'prescan_reliable', detector_regions)
+        prescan = slice_section(frame_data, obstype, 'prescan', detector_regions=detector_regions)
 
         if not return_full_frame:
             # Get the image area
-            image_data = slice_section(frame_data, obstype, 'image', detector_regions)
-            image_dq = slice_section(frame_dq, obstype, 'image', detector_regions)
-
-            # Special treatment for 3D error array
-            image_err = []
-            for err_slice in frame_err:
-                image_err.append(slice_section(err_slice, obstype, 'image', detector_regions))
-            image_err = np.array(image_err)
-
-            # Get the part of the prescan that lines up with the image
-            i_r0 = detector_areas[obstype]['image']['r0c0'][0]
-            p_r0 = detector_areas[obstype]['prescan']['r0c0'][0]
-            i_nrow = detector_areas[obstype]['image']['rows']
-            al_prescan = prescan[(i_r0-p_r0):(i_r0-p_r0+i_nrow), :]
+            if use_imaging_area: 
+                image_data = imaging_slice(obstype, frame_data, detector_regions=detector_regions)
+                image_dq = imaging_slice(obstype, frame_dq, detector_regions=detector_regions)
+
+                image_err = []
+                for err_slice in frame_err:
+                    image_err.append(imaging_slice(obstype, err_slice, detector_regions=detector_regions))
+                image_err = np.array(image_err)
+
+                prows, _, r0c0 = imaging_area_geom('SCI',detector_regions=detector_regions)
```

Comment:
Instead of 'SCI', use `obstype`.  This function gets the area right even if using 'ENG'.

Reply:
done!